### PR TITLE
feat: log warning in case of finding spring.jpa properties configuration

### PR DESCRIPTION
**Pull request by aureamunoz**
Related to #6192

I'm working on the test but maybe can have already a feedback @geoand @gsmet

### CODE REVIEW

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: We don't do start imports.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Same here.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: This will be null if you have a `spring.jpa.` property that is not defined in your map.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Maybe let's make it an entirely separate build step?

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: This is not used anymore.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: This regexp is incorrect and should be compiled. It should be `spring\.jpa\..*`. I'm not sure if you want to do that or specifically match the elements of the map.

If you do that, then you need an alternative error message in case the element is not in the map.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]:
```suggestion
                LOGGER.warn("Quarkus does not support the " + springProperty + "."
```

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]:
```suggestion
                        + " You may try to use the Quarkus equivalent : " + springJpaPropertiesMap.get(springProperty));
```

**Review by gsmet — COMMENTED** [review]
Thanks.

I added a couple of suggestions.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: You can configure your IDE to not automatically replace multiple imports with star imports

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (geoand) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Maybe a more prescriptive name would help?

**Review by gsmet — DISMISSED** [review]
Apparently, I missed the `Request changes` checkbox the first time. Adding a review to be extra sure we don't merge that one by accident.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Ah, yes!! I knew it :woman_facepalming: I was a bad configuration in my IDE. Now it's ok.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Do you mean a `@BuildStep` method? Why? I would make it in a separate `private` method but this actions don't `consume` either `produce` any items for others `steps` so I don't see why make it in a `@BuildStep`, makes sense?

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Do you mean the name of the `springJpaPropertiesMap`? 
WDYT about `springJpaToQuarkusOrmPropertiesMap`?

Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Yeah, that's what I had in mind :)

Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: thank you @geoand

**Comment by aureamunoz** [comment]
Hi, I added an IT with maven invoker plugin as @geoand suggested. I struggled with a problem of groovy version and a mix of maven plugin but I think now it's ok, at least the test. I will maybe refactor this https://github.com/quarkusio/quarkus/pull/6291#discussion_r360846745 but not sure what @gsmet means. Once you approve I will squash the commits.

File: integration-tests/spring-data-jpa/src/it/spring-configuration/pom.xml
Comment (geoand) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/pom.xml]: Maybe remove all the commented out lines?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy
Comment (geoand) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy]: +1

File: integration-tests/spring-data-jpa/src/it/spring-configuration/pom.xml
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/pom.xml]: sure!

File: integration-tests/spring-data-jpa/pom.xml
Comment (geoand) [inline integration-tests/spring-data-jpa/pom.xml]: Is this done to avoid the issue with multiple groovy versions?

File: integration-tests/spring-data-jpa/pom.xml
Comment (geoand) [inline integration-tests/spring-data-jpa/pom.xml]: I don't think we need this one, right?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm
Comment (geoand) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm]: We don't need this file, do we?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.native
Comment (geoand) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.native]: Or this one?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/resources/META-INF/resources/index.html
Comment (geoand) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/resources/META-INF/resources/index.html]: Or this one

File: integration-tests/spring-data-jpa/pom.xml
Comment (geoand) [inline integration-tests/spring-data-jpa/pom.xml]: Are we using this file?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm]: No, removed

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.native
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.native]: no :+1:

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/resources/META-INF/resources/index.html
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/resources/META-INF/resources/index.html]: yep :+1:

File: integration-tests/spring-data-jpa/pom.xml
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/pom.xml]: I have a very weird error when trying to remove this configuration so I prefer keep like this

File: integration-tests/spring-data-jpa/pom.xml
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/pom.xml]: I have a very weird error when trying to remove this configuration so I prefer keep like this

File: integration-tests/spring-data-jpa/pom.xml
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/pom.xml]: I have a very weird error when trying to remove this configuration so I prefer keep like this

File: integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm
Comment (geoand) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/src/main/docker/Dockerfile.jvm]: Thanks

File: integration-tests/spring-data-jpa/pom.xml
Comment (geoand) [reply integration-tests/spring-data-jpa/pom.xml]: OK, let's keep it for the time being

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: @gsmet any follow-up on @aureamunoz's comment?

Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: I've just seen your comment @geoand :woman_facepalming:
@gsmet did you see my comment? WDYT?

**Comment by geoand** [comment]
@aureamunoz can you please rebase onto the latest master?

Thanks

**Comment by aureamunoz** [comment]
done! Thank you @gsmet and @geoand :-)

**Comment by geoand** [comment]
@aureamunoz sorry this one slipped through the cracks!
Can you please rebase onto master just to be sure that nothing changed in the meanitime?

Thanks

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
@gsmet OK with this?

**Comment by aureamunoz** [comment]
Done @geoand, rebase onto master :-)

**Comment by geoand** [comment]
Thanks @aureamunoz

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: This should be static final.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: I don't understand why you use this weird thing? Why not use `pattern.matcher(...).matches()` below?

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: I think it would probably look better if we were using an indented list like what we do in https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/steps/ReflectiveHierarchyStep.java#L76 .

This pattern is also used in ArC and other similar things.

Comment (gsmet) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: That would do something like:
```
WARN ... Quarkus does not support the following Spring Boot configuration properties:
    - spring.jpa.show-sql: it should be replaced by quarkus.hibernate-orm.log.sql
    - spring.jpa.data.hibernate.dialect
```

File: integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy
Comment (gsmet) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy]: `Quarkus equivalent one : quarkus.hibernate-orm.log.sql.spring.jpa.show-sql`

The property looks incorrect?

File: integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy
Comment (gsmet) [inline integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy]: Why don't we have the equivalent here?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added some comments and the assertions in the test look wrong.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: IIRC the `Matcher matcher = pattern.matcher("");` statement initialize a matcher and I use it in the loop: `propertyNames.stream().filter(s -> matcher.reset(s).matches()).collect(toList())` to avoid to have a new Matcher instance every time. The `reset` method resets the matching state internally in the Matcher and makes the Matcher search through the CharSequence passed as parameter. Make sense @gsmet ?

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Ok, no problem, I didn't know it. I'm gonna do like this.

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: I'm just totally unsure this is worth it.

Compiling the regexp would be far more beneficial IMHO. I have never seen this `reset()` pattern and I'm not sure adding to the complexity is worth it in this case as I'm unsure there is a big win.

Comment (gsmet) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: But I won't fight for it if you think it's really better. The other issues concern me more.

Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: I don't remember anymore why I did it like this so ok, let's do it as you propose.

File: integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy
Comment (aureamunoz) [reply integration-tests/spring-data-jpa/src/it/spring-configuration/verify.groovy]: We have it. I would like to test the case of a Spring Boot property that we have not and take this one but it doesn't really exist. I've done it with a truly one `spring.jpa.open-in-view` for which we don't have a quarkus equivalent.

**Comment by aureamunoz** [comment]
I've just pushed the changes.Thanks for the review @gsmet, it worthed, there were a few things to fix :sweat_smile: Could you please re-check @gsmet ? or @geoand

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (Sanne) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: hi, just curious: why adding them all to a Map? At very least it could be a constant map (static).

One step further, I guess personally I'd not have used a Map at all, just check if any configuration key is equals-to-constant. No need for extensive use of lambdas when a switch on string can do :)

Comment (geoand) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Very good points @Sanne!

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (gsmet) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]:
```suggestion
                    "Quarkus does not support the following Spring Boot configuration properties:%n%s",
```
No space before colons in English :).

Comment (Sanne) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: A space after would be nice!

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Sometimes I do over engineering :woman_facepalming: :grin: Yes, good points @Sanne thank you. What about this try? Let me know @geoand @gsmet

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Done!

**Comment by aureamunoz** [comment]
What about this try? Let me know @Sanne @geoand @gsmet

**Review by geoand — APPROVED** [review]
LGTM

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java
Comment (geoand) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: It might be better to be use a StringBuilder here, but it's not terribly important

Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java]: Yes, I hesitated

**Comment by aureamunoz** [comment]
Hi, there is something to do from my side?

**Comment by geoand** [comment]
It's fine from my side. @gsmet can comment further if he thinks more changes need to be made

**Review by gsmet — APPROVED** [review]
Let's merge it.

My personal opinion is that the Map would have been more future proof when we will add more but I suppose we can redo it later when we add more.

**Comment by gsmet** [comment]
Thanks!

**Pull request merged**
